src/resources/js/views/category-color-selector.js

```diff
@@ -172,42 +179,73 @@ tribe.events.categoryColors.categoryPicker = ( function() {
 	};
 
 	/**
-	 * Adjusts dropdown position to prevent overflow.
-	 * Ensures the dropdown stays within the viewport horizontally and vertically.
+	 * Adjusts dropdown position to prevent overflow and ensures it stays within the viewport.
+	 * Anchors to the left or right of the picker depending on screen position, and retries once if needed.
 	 *
 	 * @since TBD
-	 * @param {HTMLElement} picker
-	 * @param {HTMLElement} dropdown
-	 * @return {void}
+	 * @param {HTMLElement} picker The picker element that triggers the dropdown.
+	 * @param {HTMLElement} dropdown The dropdown element to position.
+	 * @param {boolean} [retry=false] Whether this is a retry attempt.
 	 */
-	const adjustDropdownPosition = (picker, dropdown) => {
-		// Reset previous adjustments
-		dropdown.style.left = '';
-		dropdown.style.top = '';
-
-		const { left, right, top, bottom } = dropdown.getBoundingClientRect();
-		const padding = 8; // px, to avoid touching the edge.
-		const paddedViewportWidth = window.innerWidth - padding;
-		const paddedViewportHeight = window.innerHeight - padding;
-
-		// Horizontal adjustment
-		if ( right > paddedViewportWidth ) {
-			dropdown.style.left = `-${right - paddedViewportWidth}px`;
-		} else if ( left < padding ) {
-			dropdown.style.left = `${padding - left}px`;
+	const adjustDropdownPosition = ( picker, dropdown, retry = false ) => {
+		if ( ! dropdown.isConnected || ! dropdown.offsetParent ) {
+			return;
+		}
+
+		// Ensure picker is positioned relative for absolute dropdown anchoring
+		if ( getComputedStyle( picker ).position === 'static' ) {
+			picker.style.position = 'relative';
+		}
+
+		// Reset dropdown styles
+		Object.assign( dropdown.style, {
+			left: '',
+			right: '',
+			top: '',
+			position: 'absolute',
+		} );
+
+		const { left } = picker.getBoundingClientRect();
+		const vw = window.innerWidth;
+		const verticalOffset = picker.offsetHeight;
+
+		// Anchor based on proximity to screen edge
+		if ( left > vw / 2 ) {
+			dropdown.style.right = '0px';
+		} else {
+			dropdown.style.left = '0px';
 		}
 
-		// Vertical adjustment
-		if ( bottom > paddedViewportHeight ) {
-			dropdown.style.top = `-${bottom - paddedViewportHeight}px`;
-		} else if ( top < padding ) {
-			dropdown.style.top = `${padding - top}px`;
+		dropdown.style.top = `${ verticalOffset }px`;
+
+		// Prevent vertical overflow
+		const dropdownBottom = dropdown.getBoundingClientRect().bottom;
+		const maxBottom = window.innerHeight - 8;
```

Comment:
`8` seems arbitrary. At a minimum, this value should be defined as a `const` variable that indicates its purpose, e.g. `innerPadding` (hypothetical example).